tests/utils/config_test.py

```diff
@@ -0,0 +1,165 @@
+            {
+              "default": [
+                {"host": "127.0.0.1", "port": 9200},
+                {"host": "127.0.0.1", "port": 19200}
+              ],
+              "remote_1":[
+                {"host": "10.127.0.3", "port": 9200},
+                {"host": "10.127.0.8", "port": 9201}
+              ],
+              "remote_2":[
+                {"host": "88.33.27.15", "port": 39200}
+              ]
+            },
+            config.TargetHosts(os.path.join(os.path.dirname(__file__), "resources/target_hosts_2.json")).all_hosts)
+
+
+class TestClientOptions(TestCase):
+    def test_csv_client_options_parses(self):
+        client_options_string = "use_ssl:true,verify_certs:true,ca_certs:'/path/to/cacert.pem'"
+
+        self.assertEqual(
+            {'use_ssl': True, 'verify_certs': True, 'ca_certs': '/path/to/cacert.pem'},
+            config.ClientOptions(client_options_string)()
+        )
+
+        self.assertEqual(
+            {'use_ssl': True, 'verify_certs': True, 'ca_certs': '/path/to/cacert.pem'},
+            config.ClientOptions(client_options_string).default
+        )
+
+        self.assertEqual(
+            {'default': {'use_ssl': True, 'verify_certs': True, 'ca_certs': '/path/to/cacert.pem'}},
+            config.ClientOptions(client_options_string).all_client_options
+        )
+
+
+    def test_jsonstring_client_options_parses(self):
+        client_options_string = '{"default": {"timeout": 60},' \
+            '"remote_1": {"use_ssl":true,"verify_certs":true,"basic_auth_user": "elastic", "basic_auth_password": "changeme"},'\
+            '"remote_2": {"use_ssl":true,"verify_certs":true,"ca_certs":"/path/to/cacert.pem"}}'
+
+        self.assertEqual(
+            {'timeout': 60},
+            config.ClientOptions(client_options_string).default)
+
+        self.assertEqual(
+            {'timeout': 60},
+            config.ClientOptions(client_options_string)())
+
+        self.assertEqual(
+            {'default': {'timeout':60},
+             'remote_1': {'use_ssl': True,'verify_certs': True,'basic_auth_user':'elastic','basic_auth_password':'changeme'},
+             'remote_2': {'use_ssl': True,'verify_certs': True, 'ca_certs':'/path/to/cacert.pem'}},
+            config.ClientOptions(client_options_string).all_client_options)
+
+
+    def test_json_file_parameter_parses(self):
+        self.assertEqual(
+            {'default': {'timeout':60},
+             'remote_1': {'use_ssl': True,'verify_certs': True,'basic_auth_user':'elastic','basic_auth_password':'changeme'},
+             'remote_2': {'use_ssl': True,'verify_certs': True, 'ca_certs':'/path/to/cacert.pem'}},
+            config.ClientOptions(os.path.join(os.path.dirname(__file__), "resources/client_options_1.json")).all_client_options)
+
+        self.assertEqual(
+            {'default': {'timeout':60}},
+            config.ClientOptions(os.path.join(os.path.dirname(__file__), "resources/client_options_2.json")).all_client_options)
```

Comment:
os.path.join(os.path.dirname(__file__), "resources/target_hosts_1.json")

Reply:
This was a copy & paste error. I suggest the same as for the other test case above, i.e.

```python
os.path.join(os.path.dirname(__file__), "resources", "client_options_2.json")
```